src/celengine/dsodb.cpp

```diff
@@ -238,46 +235,34 @@ DSODatabase::load(std::istream& in, const fs::path& resourcePath)
             return false;
         }
 
-        DeepSkyObject* obj = nullptr;
+        std::unique_ptr<DeepSkyObject> obj;
         if (compareIgnoringCase(objType, "Galaxy") == 0)
-            obj = new Galaxy();
+            obj = std::make_unique<Galaxy>();
         else if (compareIgnoringCase(objType, "Globular") == 0)
-            obj = new Globular();
+            obj = std::make_unique<Globular>();
         else if (compareIgnoringCase(objType, "Nebula") == 0)
-            obj = new Nebula();
+            obj = std::make_unique<Nebula>();
         else if (compareIgnoringCase(objType, "OpenCluster") == 0)
-            obj = new OpenCluster();
+            obj = std::make_unique<OpenCluster>();
 
         if (obj != nullptr && obj->load(objParams, resourcePath))
         {
-            UserCategory::loadCategories(obj, *objParams, DataDisposition::Add, resourcePath.string());
+            UserCategory::loadCategories(obj.get(), *objParams, DataDisposition::Add, resourcePath.string());
 
             // Ensure that the DSO array is large enough
-            if (nDSOs == capacity)
+            if (DSOs.size() == DSOs.capacity())
             {
                 // Grow the array by 5%--this may be too little, but the
```

Comment:
Please update the comment

Reply:
On second thoughts, this growth strategy was designed for the case where a whole bunch of DSOs were loaded from a binary file, but that never happened. Might as well use the default vector growth strategy for now, especially since this vector only holds pointer-sized objects anyway.